aspnet/fundamentals/hosting.rst

```diff
@@ -0,0 +1,296 @@
+
+.. code-block:: c#
+
+    .UseUrls("http://*:5000;http://localhost:5001;https://hostname:5002")
+
+Startup Assembly ``string``
+    Key: ``startupAssembly``. Determines the assembly to search for the ``Startup`` class (unless specified using ``WebHostBuilder.UseStartup<StartupType>``, which overrides this setting). Set using non-generic ``UseStartup`` method.
+
+.. code-block:: c#
+
+    .UseStartup("StartupAssemblyName")
+
+Web Root ``string``
+    Key: ``webroot``. If not specified, Web Root will use the Content Root path. Determines default root location for static files to be served by the app (relative to Content Root). Set using ``UseWebRoot``.
+
+.. code-block:: c#
+
+    .UseWebRoot("public")
+
+A host can use :doc:`configuration` to set its values, which may be subsequently overridden. This is specified using the ``UseConfiguration`` method.
+
+.. code-block:: c#
+  :emphasize-lines: 6
+
+  public static void Main(string[] args)
+  {
+    var config = new ConfigurationBuilder()
+      .AddCommandLine(args)
+      .AddJsonFile("hosting.json", optional: true)
+      .Build();
+
+      var host = new WebHostBuilder()
+        .UseConfiguration(config)
+        .UseKestrel()
+        .Configure(app => 
+        {
+            app.Run(async (context) => await context.Response.WriteAsync("Hi!"));
+        })
+        .Build();
+
+    host.Run();
+  }
+        
+  var host = new WebHostBuilder()    
+      .UseConfiguration(config)
+      .Build();
+
+    
+Once a host is built using ``WebHostBuilder``, it is started by calling its ``Run`` extension method, which starts the web application and blocks the calling thread until the host shutdown.
+
+.. code-block:: c#
+
+    host.Run();
+
+You can run the host in a non-blocking manner by calling its ``Start`` method:
+
+.. code-block:: c#
+
+  using (host)
+  {
+    host.Start();
+    Console.ReadLine();
+  }
+
+There is also an extension method that will start the web host and listen on urls specified:
+
+.. code-block:: c#
+
+  var urls = new List<string>() { 
+    "http://*:5000", 
+    "http://localhost:5001"
+    };
+  var host = new WebHostBuilder()
+    .UseKestrel()
+    .UseStartup<Startup>();
+  
+  using (host)
+  {
+    host.Start(urls.ToArray());
+    Console.ReadLine();
+  }
+
+When configuring services as part of setting up a host, you can add services that will then be available to the constructor or ``Configure`` method in the ``Startup`` class using :doc:`dependency-injection`. You can also configure logging as part of configuring the host, using the ``ConfigureLogging`` method (:doc:`learn more about logging <logging>`). The following example demonstrates how to configure logging and a custom ``IFormatter`` service as part of building the host, and then request and use these services from ``Startup``.
```

Comment:
delete 
You can also configure logging as part of configuring the host, using the ``ConfigureLogging`` method (:doc:`learn more about logging <logging>`).

That's obvious from the code and obscure trying to read it.

Then make logging a link to your logging article 
The following example demonstrates how to configure logging <- make that a link